classifiers/nlp/ZeroShotTFClassifier/__init__.py

```diff
@@ -0,0 +1,228 @@
+import numpy as np
+
+from typing import Optional
+
+from jina.executors.classifiers import BaseClassifier
+from jina.executors.decorators import batching, as_ndarray
+from jina.executors.devices import TFDevice
+
+
+class ZeroShotTFClassifier(TFDevice, BaseClassifier):
+    """
+    :class: `ZeroShotTFClassifier` wraps tensorflow transformers from
+        huggingface, and performs zero shot learning classification
+        for nlp data.
+    """
+
+    def __init__(
+        self,
+        pretrained_model_name_or_path: str = "distilbert-base-uncased",
+        base_tokenizer_model: Optional[str] = None,
+        pooling_strategy: str = "mean",
+        layer_index: int = -1,
+        max_length: Optional[int] = None,
+        *args,
+        **kwargs,
+    ):
+        """
+        :param pretrained_model_name_or_path: Either:
+            - a string, the `model id` of a pretrained model hosted inside a
+                model repo on huggingface.co, e.g.: `bert-base-uncased`.
+            - a path to a `directory` containing model weights saved using
+                :func: `~transformers.PreTrainedModel.save_pretrained`,
+                e.g.: `./my_model_directory/`.
+        :param base_tokenizer_model: The name of the base model to use for
+            creating the tokenizer. If None, will be equal to
+            `pretrained_model_name_or_path`.
+        :param pooling_strategy: the strategy to merge the word embeddings
+            into the chunk embedding. Supported strategies include 'cls',
+            'mean', 'max', 'min'.
+        :param layer_index: index of the transformer layer that is used
+            to create encodings. Layer 0 corresponds to the embeddings layer
+        :param max_length: the max length to truncate the tokenized
+            sequences to.
+        """
+
+        super().__init__(*args, **kwargs)
+        self.pretrained_model_name_or_path = pretrained_model_name_or_path
+        self.base_tokenizer_model = (
+            base_tokenizer_model or pretrained_model_name_or_path
+        )
+        self.pooling_strategy = pooling_strategy
+        self.layer_index = layer_index
+        self.max_length = max_length
+
+        if self.pooling_strategy == "auto":
+            self.pooling_strategy = "cls"
+            self.logger.warning(
+                '"auto" pooling_strategy is deprecated, Defaulting to '
+                ' "cls" to maintain the old default behavior.'
+            )
+
+        if self.pooling_strategy not in ["cls", "mean", "max", "min"]:
+            self.logger.error(
+                f"pooling strategy not found: {self.pooling_strategy}."
+                ' The allowed pooling strategies are'
+                '"cls", "mean", "max", "min".'
+            )
+            raise NotImplementedError
+
+    def predict(self,
```

Comment:
This interface does not respect the `interface` of our `BaseClassifier`. It should only classify fron `data`, there should not be any dependency with `target_labels`

Reply:
Changed it to be part of the keywords. For zero shot classification we do need the argument in some form as the classifier will classify based on distance.